api/web3server.go

```diff
@@ -631,42 +609,33 @@ func (svr *Web3Server) getTransactionReceipt(in interface{}) (interface{}, error
 	if err != nil {
 		return nil, err
 	}
-
-	// acquire action receipt by action hash
 	actHash, err := hash.HexStringToHash256(util.Remove0xPrefix(actHashStr))
 	if err != nil {
 		return nil, errors.Wrapf(errUnkownType, "actHash: %s", actHashStr)
 	}
-	receipt, blkHash, err := svr.coreService.ReceiptByAction(actHash)
+
+	// acquire action receipt by action hash
+	selp, blockHash, _, _, err := svr.coreService.ActionByActionHash(actHash)
```

Comment:
why need changes in `getLogs`, not the scope of this PR?

Reply:
`getTransactionFromActionInfo` is changed

Reply:
why `getTransactionFromActionInfo` is changed then? is that needed for the web3 staking change? my understanding 
 of the necessary change is:
1. other than transfer, execution, check if the txData is a wrapped staking action
2. build the staking action as we discussed